pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -82,11 +94,283 @@ type DashboardServiceImpl struct {
 	k8sclient              client.K8sHandler
 	metrics                *dashboardsMetrics
 	publicDashboardService publicdashboards.ServiceWrapper
+	serverLockService      *serverlock.ServerLockService
+	kvstore                kvstore.KVStore
+
+	dashboardPermissionsReady      chan struct{}
+	k8sDeletedDashboardsCleanupJob *k8sCleanupJob
+}
+
+type k8sCleanupJob struct {
+	stop chan struct{}
+}
+
+func (b *k8sCleanupJob) Stop() {
+	if b.stop != nil {
+		close(b.stop)
+	}
+}
+
+func (dr *DashboardServiceImpl) startK8sDeletedDashboardsCleanupJob() {
+	dr.k8sDeletedDashboardsCleanupJob = &k8sCleanupJob{
+		stop: make(chan struct{}),
+	}
+
+	go func() {
+		ticker := time.NewTicker(k8sDashboardCleanupInterval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-dr.k8sDeletedDashboardsCleanupJob.stop:
```

Comment:
Same here, rather than relying on extra mechanism to stop the execution, you can use context propagated from `Run`, i.e.:
```suggestion
			case <-ctx.Done():
```